cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -157,6 +180,11 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *p
 		return false
 	}
 
+	if gadgetLayout.Size > diskLayout.Size {
+		return fmt.Errorf("device %v (%s) is too small to fit the requested layout (%s)", diskLayout.Device,
+			bytesIEC(diskLayout.Size), bytesIEC(gadgetLayout.Size))
```

Comment:
I think using `strutil.SizeToStr` would be more consistent with other numbers we output from snapd (I looked and couldn't find any situations where we output sizes formatted in IEC units